src/lib/y2network/autoinst/config_reader.rb

```diff
@@ -44,21 +45,28 @@ def initialize(section)
       # @return [Y2Network::Config] Network configuration
       def config
         attrs = { source: :sysconfig }
-        # FIXME: implement proper readers for AutoYaST
-        attrs[:interfaces] =  interfaces_reader.interfaces
-        attrs[:connections] = interfaces_reader.connections
+        # How it works? For autoyast it reads current config and apply autoyast on top of it
+        attrs[:interfaces] = system_interfaces.interfaces
+        attrs[:connections] = system_interfaces.connections
         attrs[:routing] = RoutingReader.new(section.routing).config if section.routing
         attrs[:dns] = DNSReader.new(section.dns).config if section.dns
-        Y2Network::Config.new(attrs)
+        config = Y2Network::Config.new(attrs)
+        interfaces = InterfacesReader.new(section.interfaces).config
+        intefaces.each do |interface|
+          # add or update system configuration, this will also create all missing interfaces
+          config.add_or_update_connection_config(interface)
+        end
+
+        config
       end
 
     private
 
       # Returns an interfaces reader instance
       #
       # @return [SysconfigInterfaces] Interfaces reader
-      def interfaces_reader
-        @interfaces_reader ||= Y2Network::Sysconfig::InterfacesReader.new
+      def system_interfaces
+        @system_interfaces ||= Y2Network::Sysconfig::InterfacesReader.new
```

Comment:
We are coupling AutoYaST import process to sysconfig. Another approach could be to read the system configuration and pass that `Y2Network::Config` object, so AutoYaST can read the interfaces from it (I am not sure if, in `Mode.autoinst`, that configuration is even accessible via `Yast::Lan.system_config`).

Reply:
I am also not so sure about this approach. On other hand I found quite important to somehow match system settings and on top of it apply autoyast settings.

Reply:
Yes, but my point is that this class could get current system settings from outside.

Reply:
so maybe we should try to first get config from Lan.system_config and if failed load it?

Reply:
I'd say we should use common access to system config as we do always - AY do not need to know that it is (e.g.) sysconfig source. AY reader should also deal with differences as read from the profile and process the merge of configurations as done in old code.